src/utils/AutoSyncFileFlow.hxx

```diff
@@ -62,6 +62,13 @@ public:
       start_flow(STATE(sleep_and_call_sync));
     }
 
+    /// Discontinues the automatic calls to fsync.
+    void shutdown()
+    {
+        timer_.cancel();
```

Comment:
This is not safe to call in production code (it's clearly marked in the documentation of that call).
I recommend the following pattern:
- add a boolean variable to shutdown
- at the beginning of the sync() state check the shutdown variable, if set, do `return exit()` instead of sync.

This guarantees that after calling shutdown there will be no further fsync calls without getting into race conditions of canceling timers.

Reply:
I've switched to timer_.ensure_triggered() since this could be called from external. I've also added a Notifiable which will allow the caller to be know when the timer expires and fsync is stopped.